feat: polish application ux

logan-r opened the pull request:
### Tickets:

- HCK- #906, #899 

### List of changes:

- Re-arrange application pages, questions and make look match figma design more closely
- Apply very slight gradient background so inputs have contrast
- Small tweaks to pagiation header style (use brown font and 24px line height)
- Fix key bug in pagination header
- Proper spacing of pagination header and get rid of vertical center for applications
- Standardized application page layout and got rid of spacing between buttons
- Get rid of border on school input
- Restyled checkboxes so they have more contrast, subtitle indented and grey instead of italic
- Make dropdowns arrows point downwards
- Allowed for shorter/less-wide versions of inputs (less wide field for number picker, shorter textareas with smaller character limit)
- Fix error message not displaying for previousHackathons question
- Fix display of review page and reduce repetitive code
- Fix page overflow bug that occurred on review page when user entered a long string without spaces for long answer questions
- Fix edit application page form being really squashed
- When application page changes, scroll back to top of page

### Type of change:

Please delete options that aren't relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How did you do this?

### How to test:

### Questions:

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [ ] Listed change(s) in the Changelog
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes
- [x] Referenced all useful info (issues, tasks, etc)

### Screenshots:
![image](https://user-images.githubusercontent.com/3229837/99194978-e94df780-2750-11eb-8d85-5ef960d9aeae.png)
![image](https://user-images.githubusercontent.com/3229837/99194914-970cd680-2750-11eb-92e4-cb20c7d2bc75.png)
![image](https://user-images.githubusercontent.com/3229837/99194936-aa1fa680-2750-11eb-8441-de3ea3884c3e.png)
![image](https://user-images.githubusercontent.com/3229837/99194957-cae7fc00-2750-11eb-9f70-c047258c9563.png)

jamesxu123 commented on src/features/Application/ManageApplicationForm.tsx:
```diff
@@ -666,288 +800,149 @@ const ManageApplicationForm: React.FC<IManageApplicationProps> = (props) => {
+        <div className="container">
+          <H1 marginBottom="16px">Review</H1>
+          <div className="info-text">
+            Your responses are now saved. Please review your application before
+            submitting. You won’t be able to edit your responses later.
```
Should we move this to consts?

logan-r replied:
Yeah, good catch 👍

jamesxu123 reviewed (commented):
Looks good to me. I think the portfolio page still needs to be adjusted because it looks a little asymmetrical and unbalanced, but I can probably do that in a separate PR.

logan-r commented:
Yeah I agree, the portfolio page needs to be re-arrange or something. Let's handle that as a separate PR tho.

List of changes since last review:
* Increased spacing between input
* Changed how error messages look and are placed
* Labeled some fields as optional
* Added `0` as an option for number of previous hackathons
* Fix school autoselect dropdown position
* Hide t-shirt if in remote mode
* Changed review description text
* Fix error messages for previous hackathon so their human readable (not complaining about how it's `NaN` when empty, instead just saying required)
* Made it so education background image doesn't jump around when error message appear changing form height
